scripts/deploy-fw.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+SUPPORTED_PLATFORMS=(byt cht bdw hsw apl cnl sue icl skl kbl)
+COMPILER=gcc
+DATE=$(date +%y%m%d%H%M)
+HASH=$(git log --pretty=format:"%H" -1 | cut -c1-8)
+
+function usage()
+{
+	echo "usage: deploy-build.sh remote_ip platform [options]"
+	echo "       [-x] xcc fw"
+	echo "       Supported platforms ${SUPPORTED_PLATFORMS[@]}"
+	exit 0
+}
+
+if [ "$#" -eq 0 ]
+then
+	usage
+else
+	REMOTE_IP=$1
+	shift
+	args=$1
+	shift
+	if [ "x$args" == "x" ]; then
+		usage
+	fi
+	# check for plaform
+	for i in ${SUPPORTED_PLATFORMS[@]}
+	do
+		if [ $i == $args ]
+		then
+			PLATFORM=$i
+		fi
+	done
+	if [ "x$PLATFORM" == "x" ]; then
+		usage
+	fi
+
+	# parse the args
+	for args in $@
+	do
+		if [[ "$args" == "-x" ]]
+			then
+			COMPILER=xcc
+		else
+			usage
+		fi
+	done
+fi
+
+BUILD_DIR=build_${PLATFORM}_${COMPILER}
+FW_NAME=sof-${PLATFORM}.ri
+FW_NAME_SALT=${FW_NAME}-${DATE}-${HASH}
+LDC_NAME=sof-${PLATFORM}.ldc
+
+echo "Now deploy to $REMOTE_IP with ${BUILD_DIR}/${FW_NAME}"
+ssh $REMOTE_IP "sudo mkdir -p /lib/firmware/intel/fw; mkdir -p ~/fw"
+scp ${BUILD_DIR}/${FW_NAME} $REMOTE_IP:~/fw/${FW_NAME_SALT}
+ssh $REMOTE_IP "sudo cp ~/fw/${FW_NAME_SALT} /lib/firmware/intel/fw/;cd /lib/firmware/intel;sudo ln -sf fw/${FW_NAME_SALT} ${FW_NAME}"
+scp ${BUILD_DIR}/src/arch/xtensa/${LDC_NAME} $REMOTE_IP:~/
```

Comment:
ok, I see we are keeping old versions. Not sure why this is useful in less you are doing QA. This should be enabled by a switch and default OFF.

Reply:
This won't make any harm. This scripts should only be helpful to developer. Normal users will never build binaries from source and no need to use these scripts.